pkg/minishift/openshift/service.go

```diff
@@ -75,6 +84,10 @@ type ServiceSpec struct {
 	Weight    []string
 }
 
+type ServiceSpecMap struct {
+	S map[string]*ServiceSpec
```

Comment:
`S` this is called code smell. Please give proper member name, should be at least 3 letters.

Reply:
> S this is called code smell. 

so you will also call `i` from now on `iter`, or `iterator`? Darn, seems I need to find an alternative to call `IP`, as it is not 'at least' three long.

In short, this is not a necessity. every name should be in context. Here he used it as it gets encapsulated... perhaps a function `getSpecs()` or just `get()` would be clearer even... but sure, I would not have called it 'code smell'.

Reply:
@anjannath , this is creating more confusion and also hard to get proper name. Probably, it is good fit here

I am able to achieve the same result with following:
```
func addToServiceSpecMap(s map[string]*ServiceSpec, service ServiceWeight, routeURL, namespace, nodePort string) {
	_, ok := s[service.Name]
	if ok {
		s[service.Name].URL = append(s[service.Name].URL, genUrl(service, routeURL))
		s[service.Name].Weight = append(s[service.Name].Weight, service.Weight)
	} else {
		s[service.Name] = createServiceSpec(service, routeURL, namespace, nodePort)
	}
}

func filterAndUpdateServiceSpecs(routeSpec map[string][]ServiceWeight, namespace string, serviceNodePort map[string]string) []ServiceSpec {
	var serviceSpecs []ServiceSpec

	serviceSpecMap := make(map[string]*ServiceSpec)
	for routeURL, v := range routeSpec {
		for _, service := range v {
			if hasNodePort(serviceNodePort, service.Name) {
				addToServiceSpecMap(serviceSpecMap, service, routeURL, namespace, serviceNodePort[service.Name])
				delete(serviceNodePort, service.Name)
			} else {
				addToServiceSpecMap(serviceSpecMap, service, routeURL, namespace, "")
			}
		}
	}

	for k, v := range serviceNodePort {
		serviceSpecs = append(serviceSpecs, ServiceSpec{Namespace: namespace, Name: k, URL: nil, NodePort: v, Weight: nil})
	}
	for _, v := range serviceSpecMap {
		serviceSpecs = append(serviceSpecs, *v)
	}

	return serviceSpecs
}
```

Reply:
This was what it looked like.. before.. with a little falvour (the `genUrl` function was not there), but here we are passing around `ServiceSpecMap`, and we are doing operations on it by passing it to functions.. it was a good candidate for making a struct and adding methods.